Components/FlightControl/HDITask.cpp

```diff
@@ -0,0 +1,186 @@
+
+
+extern TIM_HandleTypeDef htim2;
+
+etl::map<RocketState, BLINK, 10> stateBlinks = etl::map<RocketState, BLINK, 10>{
+    {RS_PRELAUNCH, {2, 1000}},
+    {RS_FILL, {3, 1000}},
+    {RS_ARM, {4, 1000}},
+    {RS_IGNITION, {5, 1000}},
+    {RS_LAUNCH, {6, 1000}},
+    {RS_BURN, {7, 1000}},
+    {RS_COAST, {8, 1000}},
+    {RS_DESCENT, {9, 1000}},
+    {RS_RECOVERY, {10, 1000}},
+    {RS_ABORT, {1, 1000}}
+};
+
+/**
+* @brief Constructor for HDITask
+*/
+HDITask::HDITask():Task(HDI_TASK_QUEUE_DEPTH_OBJS)
+{
+}
+
+/**
+* @brief Initialize the HDITask
+*/
+void HDITask::InitTask()
+{
+    // Make sure the task is not already initialized
+    SOAR_ASSERT(rtTaskHandle == nullptr, "Cannot initialize flight task twice");
+
+    BaseType_t rtValue =
+        xTaskCreate((TaskFunction_t)HDITask::RunTask,
+            (const char*)"HDITask",
+            (uint16_t)HDI_TASK_STACK_DEPTH_WORDS,
+            (void*)this,
+            (UBaseType_t)HDI_TASK_RTOS_PRIORITY,
+            (TaskHandle_t*)&rtTaskHandle);
+
+    SOAR_ASSERT(rtValue == pdPASS, "HDITask::InitTask() - xTaskCreate() failed");
+}
+
+
+BLINK num = stateBlinks[RS_ABORT];
+
+
+/**
+* @brief Instance Run loop for the Flight Task, runs on scheduler start as long as the task is initialized.
+* @param pvParams RTOS Passed void parameters, contains a pointer to the object instance, should not be used
+*/
+void HDITask::Run(void * pvParams)
+{
+
+	while (1) {
+
+	        Command cm;
+
+	        //Wait forever for a command,
+	        //look forward for anything in queue, if there is , if there is anything update index, if there isn't anything do what was done before
+	        //state field, last known state
+	        if(qEvtQueue->Receive(cm)){
+	        	HandleCommand(cm);
+	        }
+	        else{
+	        	BuzzBlinkSequence(num);
+	        }
+	        //Process the command
+
+	    }
+}
+
+/**
+* @brief Handles a command
+* @param cm Command reference to handle
+*/
+void HDITask::HandleCommand(Command& cm)
+{
+    //TODO: Since this task will stall for a few milliseconds, we may need a way to eat the whole queue (combine similar eg. REQUEST commands and eat to WDG command etc)
+    //TODO: Maybe a HandleEvtQueue instead that takes in the whole queue and eats the whole thing in order of non-blocking to blocking
+
+    //Switch for the GLOBAL_COMMAND
+    switch (cm.GetCommand()) {
+    case REQUEST_COMMAND: {
+        HandleRequestCommand(cm.GetTaskCommand());
+        break;
+    }
+    case TASK_SPECIFIC_COMMAND: {
+        break;
+    }
+    default:
+        SOAR_PRINT("HDITask - Received Unsupported Command {%d}\n", cm.GetCommand());
+        break;
+    }
+
+    cm.Reset();
+}
+
+/**
+* @brief Handles a Request Command
+* @param taskCommand The command to handle
+*/
+void HDITask::HandleRequestCommand(uint16_t taskCommand)
+{
+    //Switch for task specific command within DATA_COMMAND
+    switch (taskCommand) {
+    case PRELAUNCH:
+    	SOAR_PRINT("HDI Receive PreLaunch\n");
```

Comment:
nit: remove this SOAR_PRINT or make it a general one, it's unusual to have it only for prelaunch